### PR TITLE
add field for billboard target geolocations, with unit and e2e tests

### LINKED ISSUE #19794: Basic admin UI for targeting billboards by location
## Is this feature related to a problem? Please describe.

As a billboard creator, I’d like to restrict billboard displays to a particular country or region where they are most relevant.

Therefore as a platform, we need to give admin users a way to enter locations that a billboard should target.

## Describe the solution you’d like

Full international location would benefit from a more complex UI (such as country and region dropdowns), but that would also introduce complications such as how to ship all of the options to the frontend as well as i18n considerations (e.g. localizing "region" to country-specific terms such as state, province, department, etc).

In the meantime however we can ship a simple comma-separated input field of ISO 3166-2 codes that's validated by the server. The code validation is already implemented in #19790, so we just need to hook up to it

## Definition of Done

- [ ] Admins can enter and save a list of valid ISO 3166-2 codes (whole countries inclusive) as target locations in the edit billboard form
- [ ] Invalid ISO 3166-2 codes (which temporarily includes valid codes for locations outside the US and Canada) get rejected with a clear error message

## Additional Context

[This issue comment](https://github.com/forem/forem/issues/19597#issuecomment-1634771257) detailing why we are only permitting targeting the US and Canada for now

### PR DESCRIPTION
**Pull request by dukegreene**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [X] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Following up on https://github.com/forem/forem/pull/19834, this change adds a field during the creation and updating of a Billboard, and lets users set comma-separated ISO3166-2 geolocation codes (US and Canada only) according to the validations already present on the model.

This change didn't have to be too large because the model is already good at displaying specific error messages for its failed validations. We're just adding a small helper method to ensure the typed input persists in its comma-separated format as the user resubmits and the page refreshes.

## Related Tickets & Documents

- Related Issue #19794
- Closes #19794

## QA Instructions, Screenshots, Recordings

The feature flag `billboard_location_targeting` needs to be enabled to test this functionality. In localhost, I did this by going to http://localhost:3000/admin/feature_flags/features and turning it all the way on, and I believe the Admin test account will be able to do a similar toggling in Uffizzi once I draft this PR.

With the flag enabled, whether creating or editing a billboard ("Admin -> Customization -> Display Ads"), you should now see a "Target Geolocations: field with placeholder examples of the accepted format and the countries we currently support.

Errors will generate from that field if you 
  attempt to create/update with an unsupported country's valid code e.g. "MX-CMX" or
  attempt to create/update with a supported country and invalid region e.g. "US-UHHHH"
  
With otherwise valid inputs, a string of comma separated valid inputs for geolocations e.g. "US-NV, US-TX, CA-BC, CA-NS" should allow a successful save and persist on that display ad when its Show page is loaded.

If some inputs are valid and others are invalid, only the invalid ones should show up in the flash notice about form errors.

### UI accessibility checklist
_If your PR includes UI changes, please utilize this checklist:_
- [X] Semantic HTML implemented?
- [X] Keyboard operability supported? Unrelated note to self: tabbing to the "Display to user group" and "Type" radio buttons works, but there's no visual indicator that the radio groups are currently tabbed to like there is with outlined text fields. Will see if there's already an issue for this.
- [ ] Checked with [axe DevTools](https://www.deque.com/axe/) and addressed `Critical` and `Serious` issues?
- [ ] Color contrast tested?

_For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

- [X] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [optional] Are there any post deployment tasks we need to perform?

Make sure the flipper feature is enabled for the actors who plan to test it

## [optional] What gif best describes this PR or how it makes you feel?

Gina Rodriguez from the Netflix Carmen Sandiego saying "I wanted you to find me"

### CODE REVIEW

File: app/models/display_ad.rb [hunk 1]
```diff
@@ -173,6 +173,10 @@ def exclude_article_ids=(input)
     write_attribute :exclude_article_ids, (adjusted_input || [])
   end
 
+  def comma_separated_geolocations
+    target_geolocations.map { |g| "#{g.to_iso3166}, " }.join.chomp(", ")
```
Comment (filleduchaos) [inline app/models/display_ad.rb:177]: I _think_ you can accomplish same with `.map(&:to_iso3166).join(", ")`?

**Review by filleduchaos — APPROVED** [review]
QA'd locally and works great 🎉

File: app/models/display_ad.rb (hunk 1 above)
Comment (lightalloy) [reply app/models/display_ad.rb:177]: It seems like this code belongs more to a helper than to a model :eyes: 
(as it's used to format data in the view)

**Review by lightalloy — COMMENTED** [review]
Works well.
I suggest implementing changes that @filleduchaos and me mentioned.
I would also add a couple of tests to `spec/requests/admin/billboards_spec.rb` (though it's kind of covered with e2e tests)

There is also a merge conflict because renaming display ads to billboards is in process, so need to update and resolve.

**Comment by dukegreene** [comment]
Thanks @lightalloy and @filleduchaos for the helpful suggestions! I moved the refactored method work inline and got rid of the spec for the method, and also added some request specs for context

**Review by lightalloy — APPROVED** [review]
Thanks for implementing the changes :raised_hands:

**Pull request merged**